src/stories/dropdown.stories.tsx

```diff
@@ -45,7 +42,30 @@ export default {
   },
 }
 
-const Template = (args) => {
+interface Props<T> {
+  storyName: string
+  toggle: (
+    selected: {
+      name: string
+      value: T
+      placeholder?: boolean
+    },
+    isOpen: boolean
+  ) => ReactChild
+  options: Array<{
+    value: T
+    name: string
+  }>
+  onSelect?: (selection: T) => void
+  placeholder?: string
+  renderOption?: (option: {
+    value: T
+    name: string | ReactNode
+    isSelected: boolean
+  }) => ReactNode
+}
+
+const Template = <T extends Record<string, unknown>>(args: Props<T>) => {
```

Comment:
I have never used `Record`. can you outline what it generally does and why it's required here?

Reply:
I originally wanted T to extend an empty object:
const Template = <T extends {}>(args: Props<T>) => { <------ like this 

but npm run lint threw an error there:
Don't use `{}` as a type. `{}` actually means "any non-nullish value"

I'll cite stackoverflow, and link to some documentation since they are a bit better at explaining than me! :) 

> A Record<K, T> is an object type whose property keys are K and whose property values are T. That is, keyof Record<K, T> is equivalent to K, and Record<K, T>[K] is (basically) equivalent to T.

docs link: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype